File: src/dddpy/brand/domain/brand_data.py
from dataclasses import dataclass
from typing import Optional, Dict, Any


@dataclass(frozen=True)
class CreateBrandData:
    name: str
    code: str
    description: Optional[str] = None
    logo_url: Optional[str] = None


@dataclass(frozen=True)
class UpdateBrandData:
    name: Optional[str] = None
    description: Optional[str] = None
    logo_url: Optional[str] = None
    status: Optional[str] = None

File: frontend/src/features/brands/interfaces/BrandData.ts
export interface BrandTableData {
  id: string;
  code: string;
  name: string;
  logo_url: string;
  status: string;
  description: string;
}


export interface BrandInterface {
  id: string ;
  name: string;
  code: string;
  description: string;
  logo_url: string;
  status: 'ACTIVE' | 'INACTIVE';
  created_at: string | null;
  updated_at: string | null;
}